build_tools/kickstarts/cert-gen.sh

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+# Generate certs pair for configuring TLS enabled docker daemon
+. /etc/sysconfig/docker
+
+if [ -e $DOCKER_CERT_PATH/ca.pem ]; then
+  # Certificates already generated
+  exit
+fi
+
+function randomString {
+        # if a param was passed, it's the length of the string we want
+        if [[ -n $1 ]] && [[ "$1" -lt 20 ]]; then
+                local myStrLength=$1;
+        else
+                # otherwise set to default
+                local myStrLength=8;
+        fi
+
+        dd if=/dev/urandom bs=1 2>/dev/null | tr -dc '[:alnum:]' | dd bs=1 count=$myStrLength 2>/dev/null
+}
+
+# Get a temporary workspace
+dir=`mktemp -d`
+cd $dir
+
+# Get a random password for the CA and save it
+passfile=tmp.pass
+password=$(randomString 10)
+echo $password > $passfile
+
+# Generate the CA
+openssl genrsa -aes256 -passout file:$passfile -out ca-key.pem 2048
+openssl req -new -x509 -passin file:$passfile -days 365 -key ca-key.pem -sha256 -out ca.pem -subj "/C=/ST=/L=/O=/OU=/CN=example.com"
+
+# Generate Server Key and Sign it
+openssl genrsa -out server-key.pem 2048
+openssl req -subj "/CN=example.com" -new -key server-key.pem -out server.csr
+# Allow from routable local IP
+extip=`ip addr show eth1 2>/dev/null | awk 'NR==3 {print $2}' | cut -f1 -d\/`
+if [[ -z "$extip" ]]; then
+      extip=`ip route get 8.8.8.8 | awk 'NR==1 {print $NF}'`
+fi
+extipfile=extfile.cnf
+echo subjectAltName = IP:$extip > $extipfile
```

Comment:
`subjectAltName` needs portable across different providers and populated with correct IP assigned by active provider / interface.